django/s3widget/models.py

```diff
@@ -0,0 +1,48 @@
+from typing import Any, cast
+from uuid import uuid4
+
+from django.contrib.admin.widgets import AdminFileWidget
+from django.db.models.fields.files import FieldFile, FileField
+
+from .widgets import S3AdminFileInput, S3FakeFile, S3FormFileField
+
+
+class S3FieldFile(FieldFile):
+    def save(self, name, content, save=True):
+        if not isinstance(content, S3FakeFile):
+            return super().save(name, content, save)
+
+        self.name = content.name
+        setattr(self.instance, self.field.name, self.name)
+        self._committed = True
+
+        # Save the object because it has changed, unless save is False
+        if save:
+            self.instance.save()
+
+    cast(Any, save).alters_data = True
+
+
+class S3FileField(FileField):
+    attr_class = S3FieldFile
+    description = 'A file field which is uploaded to <randomuuid>/filename.'
+
+    def __init__(self, *args, **kwargs):
+        kwargs['max_length'] = kwargs.get('max_length', 200)
+        kwargs['upload_to'] = self.uuid_prefix_filename
```

Comment:
Looking at this now, maybe it should be a call to`setdefault` instead of overriding `upload_to`.